src/modules/gimbal/output.cpp

```diff
@@ -81,6 +83,27 @@ float OutputBase::_calculate_pitch(double lon, double lat, float altitude,
 	return atan2f(z, target_distance);
 }
 
+bool OutputBase::check_and_handle_setpoint_timeout(ControlData &control_data, const hrt_abstime &now)
+{
+	bool ret = false;
+	const bool timeout = (control_data.timestamp_last_update + 2_s < now);
+	const bool type_angle = (control_data.type == ControlData::Type::Angle);
```

Comment:
Wouldn't this be for angular rate?

```suggestion
	const bool type_angle = (control_data.type == ControlData::Type::AngularRate);
```

Because if the setpoint has an angle and **and** angular rate, it should go to the angle using angular rate and then stop there.

Reply:
@MaEtUgR what about this one?

Reply:
@MaEtUgR bump...

Reply:
@julianoes Sorry, I wanted to check if it's the right type before I write anything wrong. I'm pretty sure we have the rates coming with type `Angle` and not `AngularRate` otherwise this PR would have failed my test for soling the original issue.

Your suggestion conceptually makes sense but I don't yet see why I was getting type `Angle`. Is the interface misused or just wrong somewhere else?

Reply:
I finally found the time to check this in more detail.
1. I get `ControlData::Type::Angle` because of https://github.com/PX4/PX4-Autopilot/blob/cac01339018b9beddcf669f7f2f67461c37fefd0/src/modules/gimbal/input_mavlink.cpp#L924 and https://github.com/PX4/PX4-Autopilot/blob/cac01339018b9beddcf669f7f2f67461c37fefd0/src/modules/gimbal/input_mavlink.cpp#L946 and that's why my testing worked and I think it's correct.
2. To your question `AngularRate` doesn't exist:
https://github.com/PX4/PX4-Autopilot/blob/cac01339018b9beddcf669f7f2f67461c37fefd0/src/modules/gimbal/common.h#L50-L54 are you suggesting to add it? If so I'd not do that in this pull request. In interpreted the type as distinguishing if the gimbal module (gimbal manager) needs to do anything `Neutral`, forward either absolute or relative attitude setpoint for the gimbal `Angle` or run calculations to determine the desired gimbal attitude from some region of interest `LatLon`. I saw this confirmed when checking the introduction of the enum in https://github.com/PX4/PX4-Autopilot/pull/5389/files#diff-18905b738dca6fb29929c15b8ef5710cb0a5ac2e5b855e5b04147f81618bdc75R57-R62, for some reason the comments were removed in https://github.com/PX4/PX4-Autopilot/pull/19000/files#diff-7b5e5c69dd7659417c3ac6f2acff81d4b7e832f8d803d3815c328bf43a208488R50-R53
We could change the interface to represent the difference between an attitude or angular rate being commanded but currently I see even in the MAVLink interface that it's not distinguished: https://mavlink.io/en/messages/common.html#GIMBAL_MANAGER_SET_MANUAL_CONTROL
I assume this is to also allow angular rate feed forward maybe. In any case if we were to change this it'd be a bigger interface change and for this pull request it would only slightly change the condition to check if a rate is commanded that a timeout should be applied on.

Reply:
Oh, I'm sorry. I had confused it with the Frame:

https://github.com/PX4/PX4-Autopilot/blob/a3a83c718a3d6c47577dd00e4ff75be37c55572e/src/modules/gimbal/common.h#L61-L65